### PR TITLE
GHA to produce Jar release

### PR DESCRIPTION
**Pull request by connoravo**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        id: create_release
+        uses: actions/create-release@latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ needs.get-publish-version.outputs.publish-version }}
+          release_name: Release ${{ needs.get-publish-version.outputs.publish-version }}
+          draft: false
+          prerelease: false
+      - name: Gunzip Asset Prior to Upload
```
Comment (dan-hill2802) [inline .github/workflows/main.yml]: AC says "Create github actions for each repo to publish a jar file release"

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,50 @@
...
+	@{ \
+		HOOK_DIR=.git/hooks; \
+		for hook in $(shell ls .githooks); do \
+			if [ ! -h $${HOOK_DIR}/$${hook} -a -x $${HOOK_DIR}/$${hook} ]; then \
+				mv $${HOOK_DIR}/$${hook} $${HOOK_DIR}/$${hook}.local; \
+				echo "moved existing $${hook} to $${hook}.local"; \
+			fi; \
+			ln -s -f ../../.githooks/$${hook} $${HOOK_DIR}/$${hook}; \
+		done \
+	}
```
Comment (dan-hill2802) [inline Makefile]: keep these as they were otherwise they won't get updates

File: Makefile [hunk 3]
```diff
@@ -1,16 +1,50 @@
-SHELL:=bash
-
-default: help
```
Comment (dan-hill2802) [inline Makefile]: keep the default action

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: gradle build
+        run: make local-all
+      - name: Upload app
+        uses: actions/upload-artifact@v2
+        with:
+          name: built-app
+          path: build
```
Comment (dan-hill2802) [inline .github/workflows/main.yml]: just uploading the libs dir would be better

Comment (connoravo) [reply .github/workflows/main.yml]: I think this is a more verbose way - aids understanding.

**Review by dan-hill2802 — APPROVED** [review]

**Pull request merged**